.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and deploy docs
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10']
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[extra]
+        pip install .[doc]
+    - name: Build documentation
+      run: |
+        cd doc
+        make -C . html-no-examples
+    - name: Deploy Devs
+      if: success() && github.ref == 'refs/heads/master'
+      uses: JamesIves/github-pages-deploy-action@v4
+      with:
+        ssh-key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
```

Comment:
Did you have added this secret key ?

Reply:
I did not. I am not all that familiar with the keys; I assume you can do that if we want the deployment to happen.